pkg/crc/api/api_test.go

```diff
@@ -167,6 +153,14 @@ func TestGetconfigApi(t *testing.T) {
 	}, getconfigRes)
 }
 
+func setupInMemoryConfigAndHandle(fn handlerFunc, client machine.Client, args json.RawMessage) string {
+	storage := config.NewEmptyInMemoryStorage()
+	cfg := config.New(storage)
+	cmdConfig.RegisterSettings(cfg)
+
+	return fn(client, cfg, args)
+}
```

Comment:
Not clear why you need to pass `handlerFunc()`  to this? Seems like this would be simpler as
```
func inMemoryConfigFactory() config.Config {
storage := config.NewEmptyInMemoryStorage()
	cfg := config.New(storage)
	cmdConfig.RegisterSettings(cfg)

        return cfg
}
```
and `handleRequest` becomes
```
	if handler, ok := api.handlers[req.Command]; ok {
		config := api.configFactory() // or api.newConfig() or ... - this is the field which is currently called prehook
		result = handler(api.client, config, req.Args)
	} else {
...
```

Reply:
For the `NewEmptyInMemoryStorage` there is no error returned, so this is simple, but in case of the `NewViperStorage` it returns an error, and we'll have to check that in `handleRequest` before calling the handler, so then it'll be.. 

```golang
        if handler, ok := api.handlers[req.Command]; ok {
		config, err := api.newConfig()
                if err != nil {
                       result = encodeErrorToJSON(err.Error())
                } else {
		       result = handler(api.client, config, req.Args)
                }
	} else {
```

Reply:
Seems ok this way. This also allows you to add a test for the error case if needed.

Reply:
Oh, you mean as in the previous comment https://github.com/code-ready/crc/pull/1558#discussion_r500123532 i thought you'd not like the nested ifs :)

Reply:
Indeed not a big fan of the nesting, but it seems better than the alternatives in this case.

Reply:
Made the changes!